src/main/java/org/opensearch/geospatial/ip2geo/action/UpdateDatasourceRequest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.ip2geo.action;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Locale;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.log4j.Log4j2;
+
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.support.master.AcknowledgedRequest;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.ParseField;
+import org.opensearch.core.xcontent.ObjectParser;
+import org.opensearch.geospatial.ip2geo.common.DatasourceManifest;
+
+/**
+ * Ip2Geo datasource update request
+ */
+@Getter
+@Setter
+@Log4j2
+@EqualsAndHashCode(callSuper = false)
+public class UpdateDatasourceRequest extends AcknowledgedRequest<UpdateDatasourceRequest> {
+    private static final ParseField ENDPOINT_FIELD = new ParseField("endpoint");
+    private static final ParseField UPDATE_INTERVAL_IN_DAYS_FIELD = new ParseField("update_interval_in_days");
+    private static final int MAX_DATASOURCE_NAME_BYTES = 255;
+    /**
+     * @param name the datasource name
+     * @return the datasource name
+     */
+    private String name;
+    /**
+     * @param endpoint url to a manifest file for a datasource
+     * @return url to a manifest file for a datasource
+     */
+    private String endpoint;
+    /**
+     * @param updateInterval update interval of a datasource
+     * @return update interval of a datasource
+     */
+    private TimeValue updateInterval;
+
+    /**
+     * Parser of a datasource
+     */
+    public static final ObjectParser<UpdateDatasourceRequest, Void> PARSER;
+    static {
+        PARSER = new ObjectParser<>("update_datasource");
+        PARSER.declareString((request, val) -> request.setEndpoint(val), ENDPOINT_FIELD);
+        PARSER.declareLong((request, val) -> request.setUpdateInterval(TimeValue.timeValueDays(val)), UPDATE_INTERVAL_IN_DAYS_FIELD);
+    }
+
+    /**
+     * Constructor
+     * @param name name of a datasource
+     */
+    public UpdateDatasourceRequest(final String name) {
```

Comment:
Don't you need endpoint?

Reply:
Could you elaborate more? Endpoint is not needed here.